Release/src/http/client/http_client_winhttp.cpp

```diff
@@ -1159,12 +1184,21 @@ class winhttp_client : public _http_client_communicator
             case WINHTTP_CALLBACK_STATUS_DATA_AVAILABLE :
                 {
                     // Status information contains pointer to DWORD containing number of bytes available.
-                    DWORD num_bytes = *(PDWORD)statusInfo;
+                    const DWORD num_bytes = *(PDWORD)statusInfo;
 
                     if(num_bytes > 0)
                     {
-                        auto writebuf = p_request_context->_get_writebuffer();
-                        p_request_context->allocate_reply_space(writebuf.alloc(num_bytes), num_bytes);
+                        if (p_request_context->decompressor)
+                        {
+                            // Decompression is too slow to reliably do on this callback. Therefore we need to store it now in order to decompress it at a later stage in the flow.
+                            // However, we want to eventually use the writebuf to store the decompressed body. Therefore we'll store the compressed body as an internal allocation in the request_context
+                            p_request_context->allocate_reply_space(nullptr, num_bytes);
```

Comment:
When I tried to do the decompression here, I ran into an issue where a another callback to WINHTTP_CALLBACK_STATUS_READ_COMPLETE seemed to come in before the decompression had completed, and before that decompressed data was available. 

I'm now doing the decompression in the WINHTTP_CALLBACK_STATUS_READ_COMPLETE callback instead.